target_snowflake/__init__.py

```diff
@@ -113,6 +113,9 @@ def persist_lines(config, lines, table_cache=None, file_format_type: FileFormatT
     batch_size_rows = config.get('batch_size_rows', DEFAULT_BATCH_SIZE_ROWS)
     batch_wait_limit_seconds = config.get('batch_wait_limit_seconds', None)
     flush_timestamp = datetime.utcnow()
+    archive_load_files_enabled =\
```

Comment:
Think of better way to check the config - if archive_load_files.enabled but s3 bucket isn't defined, raise exception

Reply:
Config validation is at https://github.com/transferwise/pipelinewise-target-snowflake/blob/master/target_snowflake/db_sync.py#L65 , where we can do something like this:

```
    # Check if archive load files option is using external stages
    archive_load_files = config.get('archive_load_files', {})
    if archive_load_files.get('enabled') and not config.get('s3_bucket', None):
        errors.append('Archive load files option can be used only with external s3 stages. Please define s3_bucket')
```

Please note that separating config variables with dots are not working: `db_conn.s3_bucket` and `archive_load_files.enabled`. We need to access them as nested dictionaries.